protocol/discord_importer/types.go

```diff
@@ -0,0 +1,46 @@
+package discord_importer
+
+import (
+	"time"
+)
+
+type DiscordMessageAttachment struct {
+	ID string `json:"id"`
+}
+
+type DiscordMessageAuthor struct {
+	ID            string                     `json:"id"`
+	Name          string                     `json:"name"`
+	Discriminator string                     `json:"discriminator"`
+	Nickname      string                     `json:"nickname"`
+	AvatarUrl     string                     `json:"avatarUrl"`
+	Attachments   []DiscordMessageAttachment `json:"attachments"`
+}
+
+type DiscordMessage struct {
+	ID              string               `json:"id"`
+	Type            string               `json:"type"`
+	Timestamp       time.Time            `json:"timestamp"`
+	TimestampEdited string               `json:"timestampEdited"`
+	Content         string               `json:"content"`
+	Author          DiscordMessageAuthor `json:"author"`
+}
+
+type DiscordChannel struct {
+	ID           string `json:"id"`
+	CategoryName string `json:"category"`
+	CategoryId   string `json:"categoryId"`
+	Name         string `json:"name"`
+	Description  string `json:"topic"`
+	FilePath     string `json:"filePath"`
+}
+
+type DiscordCategory struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type DiscordExportedChannel struct {
+	Channel  DiscordChannel   `json:"channel"`
+	Messages []DiscordMessage `json:"messages"`
+}
```

Comment:
Chances are, these types will be extended with additional properties as soon as I start working on converting discord messages to Waku messages.